testsupport/space.go

```diff
@@ -79,9 +79,21 @@ func CreateSpace(t *testing.T, awaitilities wait.Awaitilities, opts ...SpaceOpti
 	space := NewSpace(awaitilities, opts...)
 	err := awaitilities.Host().CreateWithCleanup(context.TODO(), space)
 	require.NoError(t, err)
-
+	space, err = awaitilities.Host().WaitForSpace(space.Name, wait.UntilSpaceHasAnyTargetClusterSet(), wait.UntilSpaceHasAnyTierNameSet())
+	require.NoError(t, err)
 	// we also need to create a MUR & SpaceBinding, otherwise, the Space could be automatically deleted by the SpaceCleanup controller
-	signup, spaceBinding := CreateMurWithAdminSpaceBindingForSpace(t, awaitilities, space, true)
+	signup, mur, spaceBinding := CreateMurWithAdminSpaceBindingForSpace(t, awaitilities, space, true)
+	// make sure that the NSTemplateSet associated with the Space was updated after the space binding was created (new entry in the `spec.SpaceRoles`)
+	// before we can check the resources (roles and rolebindings)
+	tier, err := awaitilities.Host().WaitForNSTemplateTier(space.Spec.TierName)
+	require.NoError(t, err)
+	if memberAwait, err := awaitilities.Member(space.Status.TargetCluster); err == nil {
+		// if member is `unknown` or invalid (depending on the test case), then don't try to check the associated NSTemplateSet
+		_, err = memberAwait.WaitForNSTmplSet(space.Name,
+			wait.UntilNSTemplateSetHasSpaceRoles(
+				wait.SpaceRole(tier.Spec.SpaceRoles["admin"].TemplateRef, mur.Name)))
+		require.NoError(t, err)
+	}
```

Comment:
the main idea of the `CreateSpace` function was to create the Space without doing any verification - there is a different function for checking what was provisioned in there. Also the comment of this function doesn't say anything about verification.
When creating the Space you don't have to care if there was enough capacity and if the Space was really provisioned anywhere or not. For verifying that you can use different function.